SharesCalculation.py

```diff
@@ -0,0 +1,17 @@
+import ApplicationOutput as op
+import SpouseAndParentSharesCalculation as sps
+def calculate(amount, sons, daughters, mother, father, wife):
+    if amount > 0:
+        if sons > 0 and daughters >= 0 or sons >= 0 and daughters > 0:
+            amount = sps.spouse_with_sons_or_daughters(amount, mother, father, wife)
+            total_share = sons * 2 + daughters  # Calculate the total share
+            amount_per_share = amount / total_share  # Amount per share
+            op.show(sons, daughters, amount_per_share)  # Function to display the result
```

Comment:
We send the data to other function/modules when it is really necessary. Always try to inline the stuff, and refactor it later by looking where would this make sense. We don't create separate modules/function just for the sake of their usage. There should be proper reason to do that, otherwise the sequential code always makes sense and more readable. 

This discussion might be helpful: https://softwareengineering.stackexchange.com/questions/255507/at-what-point-should-you-start-putting-functions-in-a-new-file#:~:text=For%20example%2C%20if%20the%20function,the%20file%20reading%20module%2C%20etc. Or feel free to have a look on other resources. 

Always, try to adapt to best practices. Don't do something just because you feel to do it.

Reply:
Got it,but now as i mentioned i have assemmebled modules,next time I'll care about it.